src/proteus/dswx_hls.py

```diff
@@ -2355,6 +2480,105 @@ def _save_output_rgb_file(red, green, blue, output_file,
     logger.info(f'file saved: {output_file}')
 
 
+def _compute_wtr2_with_translucent_clouds_array(
+        unmasked_interpreted_water_layer,
+        cloud_layer,
+        flag_collapse_wtr_classes=FLAG_COLLAPSE_WTR_CLASSES,
+        flag_no_snow_mask=False):
+    """
+    Generate a version of the WTR-2 layer where the
+    pixels marked with 'cloud' in the cloud layer
+    appear to have translucent clouds on top.
+
+    Parameters
+    ----------
+    unmasked_interpreted_water_layer : numpy.ndarray
+        Cloud-unmasked interpreted water layer
+        (i.e. the DSWx-HLS WTR-2 layer)
+    cloud_layer : numpy.ndarray
+        DSWx-HLS CLOUD layer
+    flag_collapse_wtr_classes : bool
+        Collapse interpreted layer water classes following standard
+        DSWx-HLS product water classes
+    flag_no_snow_mask : bool
+        True to not mask the snow pixels (i.e. they will remain
+        set to the value as in `unmasked_interpreted_water_layer`)
+        False to mask pixels marked as snow/ice in `cloud_layer`.
+        Default is False (snow pixels will masked.)
+    
+    Returns
+    -------
+    wtr2_clouds_layer : numpy.ndarray
+        Interpreted water layer with "translucent" clouds
+
+    Notes
+    -----
+    CLOUD layer classifications (as of Nov. 2022)
+    0: Not masked
+    1: Cloud shadow
+    2: Snow/ice
+    3: Cloud shadow and snow/ice
+    4: Cloud
+    5: Cloud and cloud shadow
+    6: Cloud and snow/ice
+    7: Cloud, cloud shadow, and snow/ice
+    255: Fill value (no data)
+
+    For determining pixel coloration, cloud has precedence over
+    all other classes. Only color a pixel as snow if it's not 
+    cloud/cloud-shadow/adjacent masked. (e.g. a pixel with a
+    value of 6 in the CLOUD layer would be colored as cloud.)
+    """
+
+    # Create a copy of the unmasked_interpreted_water_layer.
+    wtr2_clouds_layer = unmasked_interpreted_water_layer.copy()
+
+    if flag_collapse_wtr_classes:
+        wtr2_clouds_layer = _collapse_wtr_classes(wtr2_clouds_layer)
+
+    # Mask the pixels with only snow
+    if not flag_no_snow_mask:
+        snow_idx = (cloud_layer == 2)
+        wtr2_clouds_layer[snow_idx] = WTR_CLOUD_MASKED_SNOW
+
+    # Mask the pixels with cloud and/or cloud shadow
+    cloud_idx = ((cloud_layer == 1) | 
+                (cloud_layer == 4) |
+                (cloud_layer == 5) |
+                (cloud_layer == 3) |
+                (cloud_layer == 6) |
+                (cloud_layer == 7))
```

Comment:
What if `flag_no_snow_mask` is `True`? I think, in that case, we should mimic the WTR class before we created the snow class 8, i.e., snow and clouds are represented by a single class "cloud+snow".

Reply:
It's also better to put these conditions in order, i.e, `3` before `4`, and `5`.

Reply:
> What if flag_no_snow_mask is True? I think, in that case, we should mimic the WTR class before we created the snow class 8, i.e., snow and clouds are represented by a single class "cloud+snow".

Ah, good point. Let's check with J. Jones and S. Chan?

As-is, the code is written this way to reflect my takeaway from the conversations: if the HLS FMask's encoding of snow is deemed to be faulty, then that would be the trigger to not apply the snow mask at all to those pixels, meaning that we should completely ignore the "snow" designation and not mask them.

That said, I could easily see that the takeaway you're suggesting here is actually what they meant. Totally happy to change; very easy fix.

Good question - thank you!

> It's also better to put these conditions in order, i.e, 3 before 4, and 5.
Ah - Good call.